tasks/icinga2-Debian.yml

```diff
@@ -2,17 +2,31 @@
 
 - name: Debian - Install apt-transport-https
   become: yes
-  apt: name=apt-transport-https state=present
-  when: i2_use_repository
+  apt:
+    name: apt-transport-https
+    state: present
+  when: i2_manage_repository
 
 - name: Debian - Add Icinga 2 repository key
   become: yes
-  apt_key: url="{{ i2_apt_key }}" state=present
-  when: i2_use_repository and i2_apt_key
+  apt_key:
+    url: "{{ i2_apt_key }}"
+    state: present
+  when: i2_manage_repository and i2_apt_key
 
 - name: Debian - Add Icinga 2 repository
   become: yes
-  apt_repository: repo={{ item.repo }} state={{ item.state}}
+  apt_repository:
+    repo: {{ item.repo }}
+    state: {{ item.state}}
   with_items:
```

Comment:
This is incorrect , when you define an iteration over a list , you need to define the list (and it's internal array ) in the variables
since the state is present always in that context, there is no need to put it as a variable. and as such the definition should be as follow:
apt_repository:
repo: "{{item}}"
state: present 
with_items: "{{ i2_apt_url }}"
although this is complicating it a bit , the simple way is 
apt_repository:
repo: "{{is_apt_url}}"
state: present